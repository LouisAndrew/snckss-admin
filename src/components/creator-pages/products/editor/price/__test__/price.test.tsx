import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import Price from '..'

describe('Price element on products editor', () => {
        let temp: string = ''

        const mockPrice: number = 12
        const mockHeaderText: string = 'label'
        const mockPlaceholderText: string = 'placeholder'
        const mockHandleChange = jest.fn(
                (event: React.ChangeEvent<HTMLInputElement>) => {
                        temp = event.target.value
                }
        )

        afterEach(cleanup)
        const el: React.ReactElement = (
                <Price
                        price={mockPrice}
                        headerText={mockHeaderText}
                        placeholderText={mockPlaceholderText}
                        handleChange={mockHandleChange}
                />
        )

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        it('should render with placeholder text correctly', () => {
                const { getByPlaceholderText } = render(el)
                const input: HTMLElement = getByPlaceholderText(
                        mockPlaceholderText
                )

                expect(input).toBeInTheDocument()
        })

        it('should render with label text correctly', () => {
                const { getByLabelText } = render(el)
                const input: HTMLElement = getByLabelText(mockHeaderText)

                expect(input).toBeInTheDocument()
        })

        // test passes if input is not casted to HTMLInputElement. Why?
        it('should react to user changes correctly', () => {
                const mockUserChange: string = 'change'

                const { getByLabelText } = render(el)
                const input: HTMLElement = getByLabelText(mockHeaderText)

                fireEvent.change(input, { target: { value: mockUserChange } })

                expect(mockHandleChange).toBeCalled()
                // expect(mockUserChange).toBe(temp)
                // where is the problem here?
        })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
