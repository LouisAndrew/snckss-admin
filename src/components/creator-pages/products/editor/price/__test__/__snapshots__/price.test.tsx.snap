// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price element on products editor matches snapshot 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    htmlFor="price"
  >
    label
  </label>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        Rp. 
      </span>
    </div>
    <input
      className="form-control"
      id="price"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              SyntheticEvent {
                "_dispatchInstances": null,
                "_dispatchListeners": null,
                "_targetInst": null,
                "bubbles": null,
                "cancelable": null,
                "currentTarget": [Function],
                "defaultPrevented": null,
                "dispatchConfig": null,
                "eventPhase": null,
                "isDefaultPrevented": [Function],
                "isPropagationStopped": [Function],
                "isTrusted": null,
                "nativeEvent": null,
                "target": null,
                "timeStamp": [Function],
                "type": null,
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      placeholder="placeholder"
      type="number"
      value={12}
    />
  </div>
</div>
`;
